storage-provider/server/src/pipeline/add_piece.rs

```diff
@@ -0,0 +1,248 @@
+        deal.piece_size
+    );
+    // Find the first sector with enough space for the piece
+    let sector = state.db.iter_unsealed_sectors().find(|res| match res {
+        Ok(unsealed_sector) => {
+            let free_space = unsealed_sector.free_space();
+            tracing::debug!(sector_number = %unsealed_sector.sector_number, free_space = free_space, "Checking sector...");
+            free_space > deal.piece_size
+        },
+        // Errors return false since, well, they're not valid sectors
+        _ => false,
+    });
+
+    // If we found a sector with space, we return it, otherwise, we'll create a new one
+    // NOTE(@jmg-duarte,03/02/2025): we can't keep creating sectors forever just because they dont fit
+    // (or maybe we can) but FC keeps a limit on new sectors, I don't have a solution for this NOW
+    // but we can keep this here while this implementation develops
+    // (possible) SOLUTION: pre-check sector availability when receiving deals and decide there
+    // whether we're taking the deal or not
+    if let Some(sector) = sector {
+        // NOTE(@jmg-duarte,03/02/2025): as per our filter, errors return false, as such an error couldn't be returned
+        return sector
+            .inspect(|sector| {
+                tracing::debug!(
+                    sector_number = %sector.sector_number,
+                    "Found sector for piece!",
+                );
+            })
+            .map_err(PipelineError::from);
+    }
+
+    // NOTE(@jmg-duarte,03/02/2025): comment below no longer applies but im keeping it until
+    // we have a full implementation in place
+    // TODO(@th7nder,30/10/2024): simplification, we're always creating a new sector for storing a piece.
+    // It should not work like that, sectors should be filled with pieces according to *some* algorithm.
+    let sector_number = state
+        .db
+        .next_sector_number()
+        .map_err(|err| PipelineError::CustomError(err.to_string()))?;
+    tracing::debug!(%sector_number, "Could not find a sector for piece, creating a new one...");
+
+    let unsealed_path = state.unsealed_sectors_dir.join(sector_number.to_string());
+    let sector =
+        UnsealedSector::create(state.server_info.seal_proof, sector_number, unsealed_path).await?;
+
+    Ok(sector)
+}
+
+/// Schedule a pre commit.
+///
+/// Calls to this function *MUST NOT* be made while holding a [`MutexGuard`] for `scheduled_pre_commits`.
+///
+/// If no pre-commit task has been previously scheduled, a new one is scheduled.
+/// If an existing pre-commit task is scheduled to be executed *after* the new deadline,
+/// that task will be cancelled and replaced with the new one.
+/// Otherwise, this function is a no-op.
+#[tracing::instrument(skip_all)]
+async fn schedule_pre_commit(
+    state: Arc<PipelineState>,
+    tracker: TaskTracker,
+    sector_number: SectorNumber,
+    when: Duration,
+) {
+    let deadline = Utc::now() + when;
```

Comment:
```suggestion
    let precommit_target_time = Utc::now() + when;
```

I just don't like `deadline`, it's confusing me with those deadlines in the PoSt.